internal/api/promote_downstream_v1alpha1.go

```diff
@@ -134,9 +134,9 @@ func (s *server) PromoteDownstream(
 	createdPromos := make([]*kargoapi.Promotion, 0, len(downstreams))
 	for _, downstream := range downstreams {
 		newPromo := kargo.NewPromotion(ctx, downstream, freight.Name)
-		if downstream.Spec.PromotionMechanisms == nil { // nolint: staticcheck
+		if downstream.Spec.PromotionMechanisms == nil && downstream.Spec.PromotionTemplate != nil && len(downstream.Spec.PromotionTemplate.Spec.Steps) == 0 { // nolint: staticcheck
```

Comment:
Exceeds max length allowed by linter.

```suggestion
		// nolint: staticcheck
		if downstream.Spec.PromotionMechanisms == nil &&
			downstream.Spec.PromotionTemplate != nil && 
			len(downstream.Spec.PromotionTemplate.Spec.Steps) == 0 {
```